### PR TITLE
Update the intro to the language tour

### PR DESCRIPTION
**Pull request by kwalrath**
This intro should work both if you are and aren't opted into null safety... I hope.

Staged: https://kw-www-dartlang-1.firebaseapp.com/guides/language/language-tour_migrated

### CODE REVIEW

File: src/_guides/language/language-tour_migrated.md [hunk 1]
```diff
@@ -106,14 +109,28 @@ mind:
...
 -   Although Dart is strongly typed, type annotations are optional
     because Dart can infer types. In the code above, `number`
-    is inferred to be of type `int`. When you want to explicitly say
-    that no type is expected, use the type `Object`
-    (or `Object?` if the value can be `null`) or —
-    if you want to defer type checking until runtime —
+    is inferred to be of type `int`.
+
+-   If you enable [sound null safety][ns],
+    values can’t be null unless you say they can be.
```
Comment (munificent) [inline src/_guides/language/language-tour_migrated.md:120]: Values are what they are and `null` is just another one of them floating around in value-land. Maybe a better way to say this is, "If you enable null safety, variables can't contain null unless you say they can."

File: src/_guides/language/language-tour_migrated.md [hunk 2]
```diff
@@ -106,14 +109,28 @@ mind:
...
+    putting a question mark (`?`) at the end of its type.
+    For example, a variable of type `int?` might be an integer,
+    or it might be `null`.
+
+    {{site.alert.version-note}}
+      [Null safety][ns] was introduced in Dart 2.12.
+    {{site.alert.end}}
+
+-   When you want to explicitly say
+    that no type is expected, use the type `Object?`
```
Comment (munificent) [inline src/_guides/language/language-tour_migrated.md:131]: "no type is expected" sounds sort of like it forbids everything. How about "any type is allowed"?

**Review by munificent — APPROVED** [review]
Looks good!

File: src/_guides/language/language-tour_migrated.md (hunk 1 above)
Comment (kwalrath) [reply src/_guides/language/language-tour_migrated.md:120]: Ha, I should fix this in /null-safety too!

**Pull request merged**